lib/wslistener.js

```diff
@@ -0,0 +1,226 @@
+var EventEmitter  = require('events').EventEmitter;
+var taskcluster   = require('taskcluster-client');
+var ws            = require('websocket');
+var debug         = require('debug')('taskcluster-client:wslistener');
+var _             = require('lodash');
+var assert        = require('assert');
+var urljoin       = require('url-join');
+var slugid        = require('slugid');
+
+const READYSTATE = {
+  CONNECTING: 0,
+  OPEN:       1,
+  CLOSING:    2,
+  CLOSED:     3
+};
+
+/*
+  options : {
+    baseUrl:  ....
+  }
+*/
+
+class WSListener extends EventEmitter {
+
+  constructor (options) {
+
```

Comment:
No need for empty line at start or end of function...

But I like that you split the function into segments with some empty lines :)